internal/job/integration/executor_tester.go

```diff
@@ -334,11 +334,8 @@ func (e *ExecutorTester) ReadEnvFromOutput(key string) (string, bool) {
 func (e *ExecutorTester) RunAndCheck(t *testing.T, env ...string) {
 	t.Helper()
 
-	err := e.Run(t, env...)
-	t.Logf("Bootstrap output:\n%s", e.Output)
-
-	if err != nil {
-		t.Fatalf("BootstrapTester.Run(%q) = %v", env, err)
+	if err := e.Run(t, env...); shell.GetExitCode(err) != 0 {
+		assert.NilError(t, err, "bootstrap output:\n%s", e.Output)
```

Comment:
Can you give me a bit more colour on this change?

If the exit code of err is nonzero, assert that... err == nil?